python/ray/train/v2/tests/test_metrics.py

```diff
@@ -9,43 +9,120 @@
     WorkerMetricsCallback,
 )
 from ray.train.v2._internal.execution.context import TrainRunContext
+from ray.train.v2._internal.execution.controller.state import (
+    TrainControllerStateType,
+)
+from ray.train.v2._internal.metrics.base import EnumMetric, TimeMetric
+from ray.train.v2._internal.metrics.controller import ControllerMetrics
+from ray.train.v2._internal.metrics.worker import WorkerMetrics
 from ray.train.v2.api.config import RunConfig
 
 
 class MockGauge:
-    """
-    An Naive Mock class for `ray.util.metrics.Gauge`.
-
-    This mock class initializes the value of the gauge to 0.0 and sets the value.
-    The tags keys are not used in this mock class, just for API consistency.
-    """
+    """Mock class for ray.util.metrics.Gauge."""
 
     def __init__(self, name: str, description: str, tag_keys: tuple = ()):
-        self._value: float = 0.0
+        self._values: dict[set[str], float] = {}
 
     def set(self, value: float, tags: dict):
-        self._value = value
+        self._values[frozenset(tags.items())] = value
+
+    def get(self, tags: dict):
+        return self._values.get(frozenset(tags.items()))
 
-    def get(self):
-        return self._value
+
+class MockTrainControllerState:
+    def __init__(self, state_type):
+        self._state_type = state_type
```

Comment:
just use TrainControllerState. we're mocking out the private attribute which is a bit weird here